app/scripts/auth.js

```diff
@@ -35,14 +40,28 @@ export const isParticipant = connectedReduxRedirect({
   wrapperDisplayName: 'UserIsParticipant',
 })
 
+export const isParticipantOrFreeFestival = connectedReduxRedirect({
+  redirectAction,
+  redirectPath,
+  authenticatedSelector: state => {
+    return (
+      (state.auth.isAuthenticated || state.meta.config.festivalTicketPrice === 0) &&
+      (state.user.isParticipant || state.user.isAdmin || state.meta.config.festivalTicketPrice === 0) &&
+      (state.user.isActive || state.meta.config.festivalTicketPrice === 0)
+    )
+  },
+  wrapperDisplayName: 'UserIsParticipantOrFreeFestival',
+})
+
 export const isVisitor = connectedReduxRedirect({
   redirectAction,
   redirectPath,
+  // return true statements if festival is free
   authenticatedSelector: state => {
     return (
-      state.auth.isAuthenticated &&
-      (state.user.isVisitor || state.user.isAdmin) &&
-      state.user.isActive
+      (state.auth.isAuthenticated || state.meta.config.festivalTicketPrice === 0) &&
```

Comment:
Similar thing here, maybe we change the name to `isVisitorOrFreeFestival`?

Reply:
Ok, thanks for the pointers! That all makes sense, I'll make some changes. 

What's the best way to do it now (post PR apocalypse)? Make changes directly on the development branch?

Reply:
Best way is to start a new branch / PR again with refactors (one PR per refactoring block if that makes sense). It will be easier for us to discuss the changes there before we merge it with development then :+1: